--override syntax should support initializing pointers to empty structs

/kind feature

**1. Describe IN DETAIL the feature/behavior/change you would like to see.**

In the kOps API there are a few struct types without any fields:

* [ClassicNetworkingSpec](https://pkg.go.dev/k8s.io/kops/pkg/apis/kops?utm_source=godoc#ClassicNetworkingSpec)
* [DNSAccessSpec](https://pkg.go.dev/k8s.io/kops/pkg/apis/kops?utm_source=godoc#DNSAccessSpec)
* [ExternalNetworkingSpec](https://pkg.go.dev/k8s.io/kops/pkg/apis/kops?utm_source=godoc#ExternalNetworkingSpec)
* GCENetworkingSpec
* KopeioAuthenticationSpec
* KubenetNetworkingSpec
* KuberouterNetworkingSpec
* RBACAuthorizationSpec

All of these are referenced by pointer in other struct types. Much of the relevant kOps behavior checks for non-nil values of these pointers.

Currently the --override syntax does not support initializing these empty structs:

```diff
diff --git a/pkg/commands/set_cluster_test.go b/pkg/commands/set_cluster_test.go
index bd7dd861f5..e847cecbff 100644
--- a/pkg/commands/set_cluster_test.go
+++ b/pkg/commands/set_cluster_test.go
@@ -62,6 +62,21 @@ func TestSetClusterFields(t *testing.T) {
                                },
                        },
                },
+               {
+                       Fields: []string{
+                               "spec.api.dns=",
+                       },
+                       Input: kops.Cluster{
+                               Spec: kops.ClusterSpec{},
+                       },
+                       Output: kops.Cluster{
+                               Spec: kops.ClusterSpec{
+                                       API: &kops.AccessSpec{
+                                               DNS: &kops.DNSAccessSpec{},
+                                       },
+                               },
+                       },
+               },
                {
                        Fields: []string{"spec.kubelet.authorizationMode=Webhook"},
                        Output: kops.Cluster{
```

```
--- FAIL: TestSetClusterFields (0.00s)
    src/k8s.io/kops/pkg/commands/set_cluster_test.go:284: unexpected error from setClusterFields [spec.api.dns=]: cannot set field "spec.api.dns": unhandled type "kops.DNSAccessSpec"
```

**2. Feel free to provide a design supporting your feature request.**

It would be convenient if the syntax supported initializing these empty structs, perhaps when used with an empty value after the equals sign, as demonstrated in the test case above.

With the same test case:

```
--- FAIL: TestUnsetClusterFields (0.00s)
    /src/k8s.io/kops/pkg/commands/unset_cluster_test.go:367: unexpected error from unsetClusterFields [spec.api.dns]: field spec.api.dns not found in *Cluster
```

--- a/pkg/commands/set_cluster_test.go
+++ b/pkg/commands/set_cluster_test.go
@@ -63,6 +63,21 @@ func TestSetClusterFields(t *testing.T) {
 			},
 		},
 		{
+			Fields: []string{
+				"spec.api.dns=",
+			},
+			Input: kops.Cluster{
+				Spec: kops.ClusterSpec{},
+			},
+			Output: kops.Cluster{
+				Spec: kops.ClusterSpec{
+					API: &kops.AccessSpec{
+						DNS: &kops.DNSAccessSpec{},
+					},
+				},
+			},
+		},
+		{
 			Fields: []string{"spec.kubelet.authorizationMode=Webhook"},
 			Output: kops.Cluster{
 				Spec: kops.ClusterSpec{

--- a/pkg/commands/unset_cluster_test.go
+++ b/pkg/commands/unset_cluster_test.go
@@ -63,6 +63,23 @@ func TestUnsetClusterFields(t *testing.T) {
 			},
 		},
 		{
+			Fields: []string{
+				"spec.api.dns",
+			},
+			Input: kops.Cluster{
+				Spec: kops.ClusterSpec{
+					API: &kops.AccessSpec{
+						DNS: &kops.DNSAccessSpec{},
+					},
+				},
+			},
+			Output: kops.Cluster{
+				Spec: kops.ClusterSpec{
+					API: &kops.AccessSpec{},
+				},
+			},
+		},
+		{
 			Fields: []string{"spec.kubelet.authorizationMode"},
 			Input: kops.Cluster{
 				Spec: kops.ClusterSpec{

--- a/util/pkg/reflectutils/access.go
+++ b/util/pkg/reflectutils/access.go
@@ -113,6 +113,10 @@ func setPrimitive(v reflect.Value, newValue string) error {
 		return nil
 	}
 
+	if newValue == "" && (v.Type().Kind() == reflect.Map || v.Type().Kind() == reflect.Struct) {
+		return nil
+	}
+
 	if v.Type().Kind() == reflect.Ptr {
 		val := reflect.New(v.Type().Elem())
 		if err := setPrimitive(val.Elem(), newValue); err != nil {

